src/python/pants/backend/project_info/peek.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import collections.abc
+import json
+import os
+from dataclasses import asdict, is_dataclass
+from enum import Enum
+from typing import Iterable, cast
+
+from pkg_resources import Requirement
+
+from pants.engine.addresses import Address, Addresses, BuildFileAddress
+from pants.engine.console import Console
+from pants.engine.fs import DigestContents, FileContent, PathGlobs
+from pants.engine.goal import Goal, GoalSubsystem, Outputting
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import Target, UnexpandedTargets
+
+
+class OutputOptions(Enum):
+    RAW = "raw"
+    JSON = "json"
+
+
+class PeekSubsystem(Outputting, GoalSubsystem):
+    """Display BUILD file info to the console.
+
+    In its most basic form, `peek` just prints the contents of a BUILD file. It can also display
+    multiple BUILD files, or render normalized target metadata as JSON for consumption by other
+    programs.
+    """
```

Comment:
As of yesterday, this should now be set through a class property `help`, e.g.

https://github.com/pantsbuild/pants/blob/7d8228330c77b8bd35c3411ae8375da838bdbf33/src/python/pants/vcs/changed.py#L89-L94

Note the newline characters - we want the first paragraph to still be only one sentence like this.

You can test that both `./pants help goals` and `./pants help peek` render correctly.